provider/pkg/provider/provider.go

```diff
@@ -1583,10 +1577,35 @@ func (k *kubeProvider) Diff(ctx context.Context, req *pulumirpc.DiffRequest) (*p
 				newInputs.GetNamespace(), newInputs.GetName())
 		}
 
-		for k, v := range detailedDiff {
-			switch v.Kind {
-			case pulumirpc.PropertyDiff_ADD_REPLACE, pulumirpc.PropertyDiff_DELETE_REPLACE, pulumirpc.PropertyDiff_UPDATE_REPLACE:
-				replaces = append(replaces, k)
+		// Remove any ignored changes from the computed diff.
+		for _, ignore := range req.IgnoreChanges {
+			ignorePath, err := resource.ParsePropertyPath(ignore)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse ignore path: %w", err)
+			}
+			for p := range detailedDiff {
```

Comment:
Should we flip the loops so we go over detailedDiff at the top level and look through ignoreChanges in the nested loop? I assume ignore changes is a typically smaller list?

Reply:
Curious why we need to do this? I haven't looked through engine code here but I would expect the engine to ignore things from the diff response on the engine side instead?

Reply:
@viveklak There's some additional context in https://github.com/pulumi/pulumi-kubernetes/issues/2012#issuecomment-1163775580

###### <a href="https://getunblocked.com"><img src="https://avatars.githubusercontent.com/in/166219" width=20 align="center" /></a>&nbsp;&nbsp;Comment posted from [Unblocked](https://getunblocked.com/dashboard/team/b3b49c13-3243-4494-a6b0-72bbfdc15383/thread/be5d380c-f2c8-4537-a13d-5aa0af9a98de?message=404b05f4-dd42-46da-984b-aa803fe20baf).

Reply:
I guess my question was if the engine should be doing the dropping of ignored changes given it has all the information it needs in order to perform the filtering on its side? Currently the provider has the opportunity to hand off diff to the engine with a `DiffResponse_DIFF_UNKNOWN`. Does this need to be as all-or-nothing as that?

Reply:
> Curious why we need to do this? I haven't looked through engine code here but I would expect the engine to ignore relevant things from the diff response on the engine side instead?

I remember being confused why there was an ignoreChanges passed to the provider at all when I first saw, thinking the same thing. @pgavlin added it in https://github.com/pulumi/pulumi/pull/3005. The engine will have already reset the inputs map to any old values due to ignoreChanges, I'll admit I'm not sure why this is insufficient for tf/k8ssa 😕

Reply:
https://pulumi.slack.com/archives/C011EMDQ8JE/p1649841058403479 slack thread where Pat tried to explain why it's needed.

Reply:
So I'm not quite sure why this update is needed, but it's not working with the existing Diff code. The `Update` method isn't handling `ignoreChanges` at all, but it's also not doing any additional diff calculation, so perhaps it's deferring that work to the engine. Given the extra moving parts being introduced with Patch support, I'm hesitant to make major changes to the diff logic right now, so I think we should stick with this fix for the meantime.